script/LibStateDiffChecker.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {VmSafe} from "forge-std/Vm.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {console2 as console} from "forge-std/console2.sol";
+
+library LibStateDiffChecker {
+    VmSafe private constant vm = VmSafe(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    // The order of the fields in the struct is important, and must be alphabetical for some reason related to
+    // how foundry encodes the struct to and from JSON.
+    struct StorageDiffSpec {
+        address account;
+        bytes32 newValue;
+        bytes32 previousValue;
+        bytes32 slot;
+    }
+
+    struct StateDiffSpec {
+        uint256 chainId;
+        StorageDiffSpec[] storageSpecs;
+    }
```

Comment:
Some things we neglect in our state diff: nonce increments and accounts with new code (both can happen if a contract deploys a contract)

Reply:
We agreed that this is OK for now.